xbmc/cores/VideoPlayer/VideoPlayerRadioID3.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (C) 2005-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "DVDMessageQueue.h"
+#include "FileItem.h"
+#include "IVideoPlayer.h"
+#include "threads/CriticalSection.h"
+#include "threads/Thread.h"
+
+#include <memory>
+
+#include <taglib/id3v1tag.h>
+#include <taglib/id3v2tag.h>
+#include <taglib/tstringlist.h>
+
+using namespace TagLib;
+
+class CDVDRadioID3Data : public CThread, public IDVDStreamPlayer
```

Comment:
q1: why is this derived `public`from thread, not `private`? Isn't this just an implementation detail?
q2: Or should it better be modeled as "has a" vs. current "is a" approach? "Data" **being** a thread feels awkward. Same applies to `IDVDStreamPlayer`

Reply:
How would you feel about renaming the class to actually match the file name and making CThread privately inherited (as suggested)?

`class CVideoPlayerRadioID3 : public IDVDStreamPlayer, private CThread`

The derivation from IDVDStreamPlayer is common to all of these classes and needs to stay public as VideoPlayer needs that:

```
IDVDStreamPlayer* CVideoPlayer::GetStreamPlayer(unsigned int target)
{
  if(target == VideoPlayer_AUDIO)
    return m_VideoPlayerAudio;
  ....
  if (target == VideoPlayer_ID3)
    return m_VideoPlayerRadioID3;
  return NULL;
}
```

I agree the class name is poor, this honestly just started out by copying the RDS class and changing "RDS" to "ID3".

If you're OK with this, I volunteer to refactor the RDS and Teletext classes to rename them as well? Make CThread private inheritance if applicable?

Reply:
The history of CThread is that it started with public inheritance. It was discussed and agreed that composition is better than inheritance for threading, but that would require some architectural changes. So for now the recommendation (and what I use) is private inheritance.

Reply:
> If you're OK with this, I volunteer to refactor the RDS and Teletext classes to rename them as well? Make CThread private inheritance if applicable?

 Absolutely, let's do this in a follow-up PR.